pkg/network/connection.go

```diff
@@ -1133,7 +1134,16 @@ func (cc *clientConnection) connect() (event api.ConnectionEvent, err error) {
 	if addr == nil {
 		return api.ConnectFailed, errors.New("ClientConnection RemoteAddr is nil")
 	}
-	cc.rawConnection, err = net.DialTimeout(cc.network, cc.RemoteAddr().String(), timeout)
+
+	dialer := &net.Dialer{
+		Timeout:   timeout,
+	}
+
+	if cc.mark != 0 {
+		dialer.Control = SockMarkControl
```

Comment:
maybe we can use closure to simplify logic.

Reply:
passing error through closure